src/js/media-controller.js

```diff
@@ -380,40 +418,76 @@ class MediaController extends MediaContainer {
   }
 }
 
-/*
-  Loop through child nodes and set the media[State] on every child.
-*/
-function propagateMediaState(nodeList, stateName, val) {
-  Array.from(nodeList).forEach(child => {
-    // All elements we care about at least have an empty children list (i.e. not <style>)
-    if (!child.children) return;
+const MEDIA_CONTROLLER_ATTRIBUTES = Object.values(MediaUIAttributes);
 
-    const childName = child.nodeName.toLowerCase();
+const getMediaControllerAttributesFrom = (child) => {
+  const { constructor: { observedAttributes } } = child;
+  const mediaChromeAttributesList = child?.getAttribute?.(MediaUIAttributes.MEDIA_CHROME_ATTRIBUTES)?.split?.(/\s+/);
+  if (!Array.isArray(observedAttributes || mediaChromeAttributesList)) return [];
+  return (observedAttributes || mediaChromeAttributesList).filter(attrName => MEDIA_CONTROLLER_ATTRIBUTES.includes(attrName));
+}
 
-    // Don't propagate into media elements, UI can't live in <video>
-    // so just avoid potential conflicts
-    if (child.slot === 'media') return;
-
-    function setAndPropagate() {
-      // Only set if previously defined, at least as null
-      // This is how element authors can tell us they want to
-      // receive these state updates
-      if (typeof child[stateName] !== 'undefined') {
-        child[stateName] = val;
-      }
+const isMediaControllerElement = (child) => !!getMediaControllerAttributesFrom(child).length;
```

Comment:
`isMediaControllerElement` is a little confusing of a name because we have a Media Controller Element. Instead of "media controller element/attribute" maybe we use "media UI element/attribute"?